### PR TITLE
Computed options

### PR DESCRIPTION
**Pull request by xcambar**
This is the follow up PR for https://github.com/offirgolan/ember-cp-validations/pull/229.

On top of the work done by @offirgolan in the branch `computed-options`, this PR updates the validators to take advantage of computed properties.

### CODE REVIEW

**Comment by xcambar** [comment]
@offirgolan Can you review and try to track down the failing tests, please? FYI, they pass correctly with Chrome, they just fail with PhantomJS (locally and in Travis, tried with 1.9 and 2.1).

For the sake of my own understanding, could you as well describe a use case (ideally as a test) in which the initial PR (#229) was not enough and your additions are required? Maybe because it's Sunday and my brains are not functioning properly, but I couldn't figure it out by myself 😛 .

**Comment by xcambar** [comment]
> ... and try to track down...

I meant that as "I did not succeed in having the tests pass, despite my best intentions, your help is more than welcome here" ;)

**Comment by offirgolan** [comment]
@xcambar this feature will take time. I was hoping you would create a PR from your branch to my branch `computed-options`, not into master. I need some more time to work on this and make sure everything is well tested. 

Closing this. Please create this PR targeting `computed-options`

File: addon/validations/factory.js [hunk 1]
```diff
@@ -430,7 +430,7 @@ function createTopLevelPropsMixin(validatableAttrs) {
 function getCPDependentKeysFor(attribute, validations, owner) {
   let dependentKeys = validations.map(validation => {
     const type = validation._type;
-    const options = validation.options;
+    const options = getWithDefault(validation, 'options', Ember.Object.create());
```
Comment (offirgolan) [inline addon/validations/factory.js]: validation.options is fine here. This is dealing with the original validations pojo passed in by the user.

**Pull request closed without merging**